From 3157555d0a50fbe3002354d5d2487eb8284dee64 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:49 -0400
Subject: [tools] perf/hists: Move column length calculation out of hists__inc_stats()

Message-id: <1410181752-25631-165-git-send-email-jolsa@redhat.com>
Patchwork-id: 91792
O-Subject: [PATCH RHEL7.1 BZ1134356 164/307] perf hists: Move column length calculation out of hists__inc_stats()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit ae993efc9c6bd109b027d2799a442892067e9230
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Apr 24 16:25:19 2014 +0900

description
===========
It's not the part of logic of hists__inc_stats() so it'd be better to
move it out of the function.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-diff.c b/tools/perf/builtin-diff.c
index 0e46fa1..c9cc771 100644
--- a/tools/perf/builtin-diff.c
+++ b/tools/perf/builtin-diff.c
@@ -587,6 +587,9 @@ static void hists__compute_resort(struct hists *hists)
 
   insert_hist_entry_by_compute(&hists->entries, he, compute);
   hists__inc_stats(hists, he);
+
+  if (!he->filtered)
+   hists__calc_col_len(hists, he);
  }
 }
 
diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index 12d6c1b..f5b388e 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -320,7 +320,6 @@ static struct hist_entry *hist_entry__new(struct hist_entry *template)
 void hists__inc_stats(struct hists *hists, struct hist_entry *h)
 {
  if (!h->filtered) {
-  hists__calc_col_len(hists, h);
   hists->nr_non_filtered_entries++;
   hists->stats.total_non_filtered_period += h->stat.period;
  }
@@ -687,6 +686,9 @@ void hists__output_resort(struct hists *hists)
 
   __hists__insert_output_entry(&hists->entries, n, min_callchain_hits);
   hists__inc_stats(hists, n);
+
+  if (!n->filtered)
+   hists__calc_col_len(hists, n);
  }
 }
 
-- 
1.7.1